crates/saya/core/src/lib.rs

```diff
@@ -407,3 +560,48 @@ impl From<starknet::providers::ProviderError> for error::Error {
         Self::KatanaClient(format!("Katana client RPC provider error: {e}"))
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum SayaMode {
```

Comment:
Isn't this also declared in the `shard.rs` file? If we, cleanup could be great. 👍

Reply:
`shard.rs` uses a wrapper struct: ```pub struct SayaModeArg(pub SayaMode);```.
We need to add parser trait implementations, and no reason to include it in the `saya_core`